docs/ocis/adr/0011-global-url-format.md

```diff
@@ -0,0 +1,210 @@
+* Bad, because URLs reveal no context to users
+
+### Path based URLs
+
+There is a customized ownCluod instance that uses path only based URLs:
+
+| URL | comment |
+|-|-|
+| `https://<host>/apps/files/?dir=/&` | root of the currently logged in user |
+| `https://demo.owncloud.com/apps/files/?dir=/&` | root of the currently logged in user |
+| `https://demo.owncloud.com/apps/files/?dir=/path/to/resource&` | sub folder `/path/to/resource` |
+
+* Good, because the URLs reveal the full path context to users
+* Good, because powerusers can navigate by updating the path in the url
+* Bad, because the bookmarks break when someone renames a folder in the path
+* Bad, because there is no id that can be used as a fallback lookup mechanism
+* Bad, because URLs might leak too much context (parent folders of shared files)
+
+### Space based URLs
+
+| URL | comment |
+|-|-|
+| `https://<host>/#/s/<space_id>(/<relative/path>)(?id=<resource_id>)` | the pattern, relative `path` and `resource_id` are optional |
+| `https://demo.owncloud.com/#/s/b78c2044-5b51-446f-82f6-907a664d089c:194b4a97-597c-4461-ab56-afd4f5a21607` | root of a storage space, might be the currently logged in users home |
+| `https://demo.owncloud.com/#/s/b78c2044-5b51-446f-82f6-907a664d089c:194b4a97-597c-4461-ab56-afd4f5a21607/relative/path/to/resource` | sub folder `/relative/path/to/resource` in the storage with id `b78c2044-5b51-446f-82f6-907a664d089c:194b4a97-597c-4461-ab56-afd4f5a21607`, works ***only*** if path still exists  |
+| `https://demo.owncloud.com/#/s/b78c2044-5b51-446f-82f6-907a664d089c:194b4a97-597c-4461-ab56-afd4f5a21607/relative/path/to/resource?id=ba4c1820-df12-11eb-8dcd-ff21f12c1264:beb78dd6-df12-11eb-a05c-a395505126f6` | sub folder `/relative/path/to/resource` in the storage with id `b78c2044-5b51-446f-82f6-907a664d089c:194b4a97-597c-4461-ab56-afd4f5a21607`, lookup can fall back to the `id` |
+
+{{< hint >}}
+* `/#` is used by the current vue router.
+* `/s` denotes that this is a space url.
+* `<space_id>` and `<resource_id>` both consist of `<storage_id>:<node_id>`, but the `space_id` can be replaced with a shorter id or an alias. See furthor down below.
+* `<relative/path>` takes precedence over the `<resource_id>`, both are optional
+{{< /hint >}}
+
+* Good, because the URLs reveal a relevant path context to users
+* Good, because everything after the `#` is not sent to the server, building the webdav request to list the folder is offloaded to the clients
+* Good, because powerusers can navigate by updating the path in the url
+* Bad, because the current ids are uuid based, leading to very long URLs where tha path component nearly vanishes between two very long strings
+* Bad, because the `#` in the URL is just a technical requirement
+* Bad, because ocis web requires a `/#/files/s` at the root of the route to distinguish the files app from other apps
+* Bad, while navigating using the WebUI, the URL has to be updated whenever we change spaces.
```

Comment:
Right. This is confusing for the user. Especially when we decided to always update the spaceid to the most recent opened space. This is a double edge sword. On the one hand it potentially shortens URLs; say we have two spaces: `spaceA`, and `spaceA/spaceB`, `pwd=spaceA/` the url would show the current space's root: `/#/spaceA/` now the user `cd` into `spaceB/ and the URL updates to: `/#/spaceB`, where is `spaceA`? This might seem counter-intuitive for the end-user.

Reply:
It is confusing but it might be a feature when it is used to allow adding context to shares. I propose allowing users to copy the share link, similar to copying the curren public link: `https://demo.owncloud.com/shares/demo/foo bar baz?id=c6e0f322-eea0-4fcd-b8a0-91d535200124:b1fb4b2a-4134-43ef-a5b9-d5985abc3d1c` is a perfectly valid bookmark. In `/shares/demo/foo bar baz` `foo bar baz` could be the _name_ of the share, regardless of the shared folder. It could be a token, as for public links, or it could be the name of the originally shared resource if the user did not type in a custom name. To prevent collisions we can append a number in brackets `foo bar baz (2)`. This won't even change when `foo bar baz` (the first one) is unshared.